File: Selenium/Activities/activity10.java
package activities;

import io.github.bonigarcia.wdm.WebDriverManager;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;

import java.time.Duration;

public class activity10 {
    public static void main(String[] args) {
        // Set up Firefox driver
        WebDriverManager.firefoxdriver().setup();
        // Create a new instance of the Firefox driver
        WebDriver driver = new FirefoxDriver();
        // Create the Wait object
        WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(10));

        // Open the page
        driver.get("https://training-support.net/selenium/DynamicControls");
        // Print the title of the page
        System.out.println("Home page title: " + driver.getTitle());

        // Find the checkbox
        WebElement checkbox = driver.findElement(By.xpath("//input[@id='checkbox']"));
        // Find the toggle button and click it
        WebElement checkboxToggle = driver.findElement(By.xpath("//button[contains(text(),'Toggle Checkbox')]"));
        checkboxToggle.click();
        // Wait for the checkbox to disappear
        wait.until(ExpectedConditions.invisibilityOfElementLocated(By.xpath("//input[@id='checkbox']")));
        System.out.println("Checkbox is displayed: " + checkbox.isDisplayed());
        // Click the button again
        checkboxToggle.click();
        // Wait for the element to appear
        wait.until(ExpectedConditions.visibilityOfElementLocated(By.xpath("//input[@id='checkbox']")));
        System.out.println("Checkbox is displayed: " + checkbox.isDisplayed());
        // Click the checkbox
        driver.findElement(By.xpath("//input[@id='checkbox']")).click();

        // Close the browser
        driver.close();
    }
}

File: Selenium/Activities/activity10.py
from selenium.webdriver.common.by import By
from selenium import webdriver
from selenium.webdriver.firefox.service import Service as FirefoxService
from selenium.webdriver.support import expected_conditions
from selenium.webdriver.support.wait import WebDriverWait
from webdriver_manager.firefox import GeckoDriverManager
driver = webdriver.Firefox(service=FirefoxService(GeckoDriverManager().install()))
wait = WebDriverWait(driver, 10)
# Navigate to URL
driver.get("https://training-support.net/selenium/DynamicControls")
print("Page title is: ", driver.title)
checkbox = driver.find_element(By.XPATH, "//input[@id='checkbox']")
checkbox_toggle = driver.find_element(By.XPATH, "//button[contains(text(),'Toggle Checkbox')]")
print("Checkbox is visible: ", checkbox.is_displayed())
checkbox_toggle.click()
wait.until(expected_conditions.invisibility_of_element(checkbox))
print("Checkbox is visible: ", checkbox.is_displayed())
